internal/api/middleware/storeAPIVersion.go

```diff
@@ -0,0 +1,22 @@
+package middleware
+
+import (
+	"net/http"
+	"playbook-dispatcher/internal/common/utils"
+	"strings"
+)
+
+func StoreAPIVersion(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.Contains(r.URL.Path, "/v1/") {
+			ctx := utils.WithApiVersion(r.Context(), "v1")
+			next.ServeHTTP(w, r.WithContext(ctx))
+		} else if strings.Contains(r.URL.Path, "/v2/") {
+			ctx := utils.WithApiVersion(r.Context(), "v2")
+			next.ServeHTTP(w, r.WithContext(ctx))
+		} else {
+			ctx := utils.WithApiVersion(r.Context(), "v1")
+			next.ServeHTTP(w, r.WithContext(ctx))
+		}
```

Comment:
What do you think about the following refactoring?

Bascially, just factor out the setting of the version within the context and the next.ServeHTTP() call to the bottom of the function and only set the api version to "v2" if we find "v2" in the url.

```go
                apiVersion := v1
		if strings.Contains(r.URL.Path, "/v2/") {
                  apiVersion = v2
		}
		ctx := utils.WithApiVersion(r.Context(), apiVersion)
		next.ServeHTTP(w, r.WithContext(ctx))
```

Reply:
I like it!  changing now!